src/test/compile-fail/gated-naked_functions.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#[naked]
+fn naked() {
+    //~^ the `#[naked]` attribute is an experimental feature
```

Comment:
Fair enough. Travis failure suggests it needs to be placed on the attribute, though.